model/gateway/gateway.go

```diff
@@ -0,0 +1,10 @@
+package gateway
+
+import "encoding/xml"
+
+// ErrorResponse is used in gateway error response
+type ErrorResponse struct {
+	XMLName xml.Name `xml:"Error"`
+	Code    int32    `xml:"Code"`
+	Message string   `xml:"Message"`
+}
```

Comment:
This data structure can be reused uniformly in `gateway`/`ratelimiter`.

Reply:
yes, but this may cause a big change, I should do this carefully.

Reply:
TODO.

Reply:
fixed